### ISSUE TITLE
Checkpointing: Add state root in bottom-up checkpoint

### ISSUE THREAD

**Issue by adlrocha** [consensus-shipyard/ipc-agent#186]
The proof for bottom-up checkpoints is currently the key of the tipset for the epoch of the checkpoint being submitted. While this allows to follow that the history of the chain is consistent, it is a quite weak proof. To make it more robust and start moving towards a way to spawn a subnet from the state of a checkpoint, we should include in the proof the state root in the checkpoint's epoch, or both: the tipset key and the state root. The proof could be the serialization of:

```rust
struct {
  tipset: Cid
  state: Cid
}
```

**Comment by cryptoAtwill** [comment]
btw, @adlrocha I think it's also good to add a version as the first field so that it's more upgrading friendly, something like:
```
enum CheckProof {
  V1 { tipset: Cid, state: Cid }
}
```

**Comment by adlrocha** [comment]
> btw, @adlrocha I think it's also good to add a version as the first field so that it's more upgrading friendly, something like:

LGTM. Let's do this, even if we make breaking changes for versions initially until we make it to mainnet. I am thinking that if we want these proofs to be custom for subnets we may need to add some `metadata` name or unique id in the future, but version should be fine for now.

**Comment by adlrocha** [comment]
Implemented in #199
